### PR TITLE
build-pdf: only build changed folders since last commit

### PR DESCRIPTION
**Pull request by sebastiaanspeck**
I noticed the Build-step took very long. This was due to the fact that all pages-folders were build every commit, even if they didn't change. By adding the provided change in this PR, we will gather the pages-folders that were changed in the last commit and deploy those. I have tested this locally as far as I could test it locally.

This PR resolves this issue: https://github.com/tldr-pages/tldr/pull/11773#issuecomment-1862658871

### CODE REVIEW

**Review by kbdharun — APPROVED** [review]
LGTM, great work with this PR @sebastiaanspeck.

**TLDR**. I have tested all the changes by setting up tldr-bot locally in my server and testing the changes in workflow in my fork. All the suggestions/issues mentioned at https://github.com/tldr-pages/tldr/pull/11773#issuecomment-1862658871 have been addressed. I will go over them in detail below for future reference (as I didn't document them in the previous PR).

---

Currently, the PDF build script runs on every commit for all languages (even if they aren't modified the artifacts are generated and deployed to the website repo) this leads to a few issues:

1. This effectively increased our CI's average workflow runtime from 2-3 minutes to 10+ minutes (as the build PDF step took a lot of time).
2. This caused space issues in the already large website repo (6 GB+) as new PDFs were being uploaded on each commit.

@sebastiaanspeck reached out to me about it a few days ago, that he was interested in working on optimizing it and #11773 was created. It previously had some issues namely:

1. Since PDFs aren't being generated, the `mv -f` step in the deploy script failed, causing no artefacts to be uploaded (this is problematic as we exclude a few languages from the PDF generation script due to rendering issues with their respective typefaces).
2. This change meant the SHA checksum for PDF files would get removed for old PDFs and only be available for newer ones. (Since we don't need them I suggested removing PDFs from the checksum step).

---

Current PR addressed the above concerns by:

1. Updating the `mv -f` command to use `find` for more efficiency (and better compatibility).
2. Update the PDF script so that all the PDF files are still committed (with no increase in size) but now for languages not modified in a PR the build PDF doesn't build it and only builds the required PDFs, this decreases the workflow run time drastically back to the 2-4 minute mark.
3. While PDF SHA checksum is possible now with the updated script, I don't think it is necessary (I initially added it over there without thinking twice). As it is primarily used by clients like `tlrc` to verify the checksum of the fetched ZIP archive.

---

**How did I test it?**

After the last PR's action failed, I immediately reverted the change and discussed with Sebastiaan to set up a test environment with tldr-bot to test it. I did a small setup of [tldr-bot](https://github.com/tldr-pages/tldr-bot) in my server and updated the relevant configuration in my forks to test the changes. (I will be deleting the fork after this PRs merge, will archive it using Internet Archive and link it here)

## Links

- [Actions page](https://web.archive.org/web/20231220162457/https://github.com/kbdkorg/tldr-build-pdf-test/actions)
- Logs (of commits):
  - test: pdf single language (Here, I modified only a single language so only a single PDF is built)-> [logs_26.zip](https://github.com/tldr-pages/tldr/files/13729832/logs_26.zip)
  - test: excluded language (Here, I tested it with an excluded language where no PDFs are newly built) -> 
[logs_27.zip](https://github.com/tldr-pages/tldr/files/13729853/logs_27.zip)
  - Similarly, I tested it by modifying non-page files too.

This is the commit details on the website repo (fork) side:

 - [test: pdf single language](https://web.archive.org/web/20231220163436/https://github.com/kbdkorg/tldr-pages.github.io/commit/3ee6199ad95de4e2bdb5a2cc903dddff649e63e4)
 - [test: excluded language](https://web.archive.org/web/20231220163439/https://github.com/kbdkorg/tldr-pages.github.io/commit/73ed273d9bbf57b4c9d4b267acc7fbb2a6130509)

**Review by gutjuri — APPROVED** [review]
Thanks for your work guys!!

**Comment by kbdharun** [comment]
> Ready to merge or shall we wait for @sbrl review?

I think it is GTG (as we have thoroughly tested the script).

File: scripts/deploy.sh [hunk 1]
```diff
@@ -31,16 +31,15 @@ function initialize {
 function upload_assets {
   git clone --quiet --depth 1 git@github.com:${SITE_REPO_SLUG}.git "$SITE_HOME"
   mv -f "$TLDR_ARCHIVE" "$SITE_HOME/assets/"
-  mv -f "${TLDR_LANG_ARCHIVES_DIRECTORY}"/*.zip "$SITE_HOME/assets/"
+  find "$TLDR_LANG_ARCHIVES_DIRECTORY" -maxdepth 1 -name "*.zip" -exec mv -f {} "$SITE_HOME/assets/" \;
```
Comment (sbrl) [inline scripts/deploy.sh:34]: Need a single quote on `-name` here to be completely robust.

```suggestion
  find "$TLDR_LANG_ARCHIVES_DIRECTORY" -maxdepth 1 -name '*.zip' -exec mv -f {} "$SITE_HOME/assets/" \;
```

File: scripts/deploy.sh [hunk 2]
```diff
@@ -31,16 +31,15 @@ function initialize {
...
   rm -rf "$TLDR_LANG_ARCHIVES_DIRECTORY"
   cp -f "$TLDRHOME/index.json" "$SITE_HOME/assets/"
-  mv -f "${TLDR_PDF_FILES_DIRECTORY}"/*.pdf "${SITE_HOME}/assets/"
+  find "$TLDR_PDF_FILES_DIRECTORY" -maxdepth 1 -name "*.pdf" -exec mv -f {} "$SITE_HOME/assets/" \;
```
Comment (sbrl) [inline scripts/deploy.sh:37]: Same here.
```suggestion
  find "$TLDR_PDF_FILES_DIRECTORY" -maxdepth 1 -name '*.pdf' -exec mv -f {} "$SITE_HOME/assets/" \;
```

**Review by sbrl — COMMENTED** [review]
Since I was @mentioned I took a look anyway, and this seems fine! Just spotted a potential robustness issue, but it's very minor.
